react/week3/todolist/src/App.js

```diff
@@ -0,0 +1,217 @@
+                    {updateStatus ? <span>{todoDescription}</span> : <input onChange={(e) => inputChange(e)} value={todoDescription} ></input>}
+                    <span className="todo-deadline">{deadline}</span>
+                </li>
+            </div>
+            <div className="buttons-container">
+                <button className="edit-button" onClick={handleUpdateStatus} >{updateStatus ? "Edit" : "Update"} </button>
+                <button className="delete-button" onClick={() => props.deleteMethod(id)}> Delete </button>
+            </div>
+        </div>
+    )
+}
+
+CreateTodo.propTypes = {
+    description: PropTypes.string.isRequired,
+}
+
+
+function DescriptionInput(props) {
+    return (
+        <div>
+            <label>Description</label>
+            <input className="description-input" onChange={(e) => props.changeMethod(e.target, 'description')} type="text" name="description" value={props.value} autoComplete="off"></input>
+        </div>
+    )
+}
+
+function DeadlineInput(props) {
+    const [startDate, setStartDate] = useState(null);
+    const date = new Date();
+
+    return (
+        <DatePicker
+            placeholderText="Click to select a date"
+            closeOnScroll={true}
+            selected={startDate}
+            onChange={(date) => {
+                setStartDate(date)
+                props.changeMethod(date, 'date')
+            }
+            }
+            name="deadline"
+            value={props.value}
+            minDate={date}
+            openToDate={date}
+            className="date-input"
+            autoComplete="off"
+        />
+    )
+}
+
+function TodoForm(props) {
+    const [inputTodo, setInputTodo] = useState({
+        description: "",
+        deadline: ""
+    });
+    function setNewInput(inputValue, inputName) {
+        if (inputName === 'date') {
+            setInputTodo((prev) => {
+                return ({ ...prev, deadline: inputValue.getFullYear() + "-" + ((inputValue.getMonth() < 10) ? "0" : "") + (inputValue.getMonth() + 1) + "-" + ((inputValue.getDate() < 10) ? "0" : "") + inputValue.getDate() })
+            })
+        } else if (inputName === 'description') {
+            const { name, value } = inputValue;
+            setInputTodo((prev) => {
+                return ({ ...prev, [name]: value })
+            })
+        }
+    }
+
+    const { addMethod } = props
+    return (
+        <div className="add-todo-container">
+            <h3>Add a todo</h3>
+            <div className="inputs-container">
+                <DescriptionInput changeMethod={setNewInput} value={inputTodo.description} />
+                <DeadlineInput changeMethod={setNewInput} value={inputTodo.deadline} />
+                <div className="add-button-container">
+                    <button onClick={() => {
+                        addMethod(inputTodo)
+                        setInputTodo({
+                            description: "",
+                            deadline: null
```

Comment:
```suggestion
                            deadline: ""
```
the initial state is ""

  const [inputTodo, setInputTodo] = useState({
    description: "",
    deadline: "",
  });

Reply:
Thanks, this solved my problem that the Datepicker didn't reset after I added a Todo.